libs/cardano-ledger-core/src/Cardano/Ledger/Plutus/Data.hs

```diff
@@ -209,6 +210,25 @@ data Datum era
   | Datum !(BinaryData era)
   deriving (Eq, Generic, NoThunks, Ord, Show)
 
+instance Era era => MemPack (Datum era) where
+  packedByteCount = \case
+    NoDatum -> packedTagByteCount
+    DatumHash dataHash -> packedTagByteCount + packedByteCount dataHash
+    Datum binaryData -> packedTagByteCount + packedByteCount binaryData
+  {-# INLINE packedByteCount #-}
+  packM = \case
+    NoDatum -> packTagM 0
```

Comment:
The tags don't  match with the `EncCBOR` (even though `Datum` is part of `TxOut`),  so now I'm fairly certain that they're not intended to match at all.

Reply:
Yep, same here. CBOR and MemPack are unrelated

Reply:
Also there is no need why they should match up. It might look inconsistent, but in the end it doesn't matter, since two serialization methods are unrelated